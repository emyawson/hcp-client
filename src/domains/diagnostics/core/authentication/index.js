export {
  validateSession,
  signOut,
  onRedirectToLogin,
} from './authentication.actions';
export {
  getToken,
  getEc6Token,
  getIsValidSession,
  selectHasTemporaryPassword,
  selectHasLoggedOut,
} from './authentication.selectors';
export {
  LOGIN,
  LOGIN_BASE,
  FIRST_TIME,
  FIRST_TIME_BASE,
  VALIDATE_SESSION,
  VALIDATE_SESSION_BASE,
  RESET_PASSWORD,
  RESET_PASSWORD_BASE,
  SIGN_OUT,
  SIGN_OUT_BASE,
  REDIRECT_TO_LOGIN,
} from './authentication.constants';
