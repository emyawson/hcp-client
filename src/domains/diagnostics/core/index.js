export { validateSession } from './authentication';
export { createModal, MODAL_TYPES } from './modal';
export {
  fetchPatientRequest,
  selectPatient,
  selectPatientDevices,
} from './patient';
export {
  clearPatientDateRange,
  fetchPatientDateRangeRequest,
  onPatientDateRangeChange,
  selectPatientFirstMeasurementDate,
  selectPatientLastMeasurementDate,
  selectPatientStartDate,
  selectPatientEndDate,
} from './patient-date-range';
export {
  createRequestActions,
  createRequestActionTypes,
  REQUEST,
  REQUEST_ANIMATION_DELAY,
  selectOngoingRequests,
} from './request';
export {
  getThresholdsRequest,
  getTimeIntervalsRequest,
  getTrafficLightColorFromStatus,
  TRAFFIC_LIGHT_STATES,
} from './strip-delivery';
