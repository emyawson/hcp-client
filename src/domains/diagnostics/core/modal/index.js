export { createModal } from './modal.actions';
export { MODAL_ACTIONS, MODAL_TYPES } from './modal.constants';
