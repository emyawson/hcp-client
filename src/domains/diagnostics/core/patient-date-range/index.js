export {
  fetchPatientDateRangeRequest,
  onPatientDateRangeChange,
} from './patient-date-range.action';
export {
  selectPatientStartDate,
  selectPatientEndDate,
  selectPatientFirstMeasurementDate,
  selectPatientLastMeasurementDate,
} from './patient-date-range.selector';
