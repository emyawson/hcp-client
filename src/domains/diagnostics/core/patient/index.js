export { fetchPatientRequest } from './patient.action';
export { FETCH_PATIENT } from './patient.constant';
export { selectPatient, selectPatientDevices } from './patient.selector';
