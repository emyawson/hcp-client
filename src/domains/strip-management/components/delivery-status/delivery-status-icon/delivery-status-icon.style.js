import styled from 'styled-components';

import {
  borderRadius,
  spacing,
  transitions,
  transitionSpeed,
  getTrafficLightColorFromStatus,
} from 'src/core';

import { TrafficIconSpan } from '../traffic-light/traffic-light.style';

export const DeliveryStatusWrapperDiv = styled.div`
  border-radius: ${borderRadius.circle};
  border: 1px dashed ${props => getTrafficLightColorFromStatus(props.status)};
  display: inline-block;
  padding: ${spacing.two};
  margin-right: ${spacing.three};
  transition: ${transitions.default};

  ${TrafficIconSpan} {
    animation-delay: ${transitionSpeed.default};
  }
`;
