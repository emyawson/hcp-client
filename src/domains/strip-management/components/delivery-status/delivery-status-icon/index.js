export * from './delivery-status-icon.component';
