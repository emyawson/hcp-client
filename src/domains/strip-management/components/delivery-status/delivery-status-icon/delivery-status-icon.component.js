import React from 'react';

import { TRAFFIC_LIGHT_STATES } from 'src/core';

import { DeliveryStatusWrapperDiv } from './delivery-status-icon.style';

import { TrafficLight } from '../traffic-light';

export const DeliveryStatusIcon = ({
  status = TRAFFIC_LIGHT_STATES.DISABLED,
  noIcon = false,
}) => (
  <DeliveryStatusWrapperDiv status={status}>
    <TrafficLight status={status} noIcon={noIcon} size={36} />
  </DeliveryStatusWrapperDiv>
);
