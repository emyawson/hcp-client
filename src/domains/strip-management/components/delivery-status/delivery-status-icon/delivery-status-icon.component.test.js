import React from 'react';
import { shallow } from 'enzyme';

import { TRAFFIC_LIGHT_STATES } from 'src/core';

import { DeliveryStatusIcon } from './delivery-status-icon.component';

describe('Delivery status icon component', () => {
  const mockProps = {
    status: TRAFFIC_LIGHT_STATES.DELIVER,
    noIcon: true,
  };
  it('Renders correctly', () => {
    const wrapper = shallow(<DeliveryStatusIcon {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Renders with default props', () => {
    const wrapper = shallow(<DeliveryStatusIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});
