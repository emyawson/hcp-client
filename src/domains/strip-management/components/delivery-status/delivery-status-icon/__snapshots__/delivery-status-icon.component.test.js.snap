// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delivery status icon component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    noIcon={true}
    status="DELIVER"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown
        noIcon={true}
        size={36}
        status="DELIVER"
      />,
      "status": "DELIVER",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "noIcon": true,
        "size": 36,
        "status": "DELIVER",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown
          noIcon={true}
          size={36}
          status="DELIVER"
        />,
        "status": "DELIVER",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "noIcon": true,
          "size": 36,
          "status": "DELIVER",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Delivery status icon component Renders with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown
        noIcon={false}
        size={36}
        status="DISABLED"
      />,
      "status": "DISABLED",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "noIcon": false,
        "size": 36,
        "status": "DISABLED",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown
          noIcon={false}
          size={36}
          status="DISABLED"
        />,
        "status": "DISABLED",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "noIcon": false,
          "size": 36,
          "status": "DISABLED",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
